sensor: dts - convert several sensor drivers to DTS

galak opened the pull request:
Convert th02 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: th02: Convert to DTS

Convert th02 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ulfalizer commented on dts/bindings/sensor/hoperf,th02.yaml:
```diff
@@ -0,0 +1,11 @@
+title: TH02 Temperature and Humidity sensor
+
+description: |
+    This is a representation of TH02 Temperature and Humidity sensor
```
Could replace with

    TH02 Temperature and Humidity sensor. See datasheet at...

Or whatever other information might be useful.

Would be nice if no more "this is a (base) representation of..." could be added. :)

galak replied:
Fixed.

ulfalizer commented:
> **Some checks failed. Please fix and resubmit.**
> 
> ## License issues
> In most cases you do not need to do anything here, especially if the files
> reported below are going into ext/ and if license was approved for inclusion
> into ext/ already. Fix any missing license/copyright issues. The license
> exception if a JFYI for the maintainers and can be overriden when merging the
> pull request.
> 
> * samples/sensor/th02/boards/frdm_k64f.overlay missing copyright.
> 
> _Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages._

Related PR: https://github.com/zephyrproject-rtos/ci-tools/pull/105

galak committed:
sensor: tmp112: Convert to DTS

Convert tmp112 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: tmp007: Convert to DTS

Convert tmp007 sensor driver to utilize device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: sx9500: Convert to DTS

Convert sx9500 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ulfalizer reviewed (approved):
LGTM

galak committed:
sensor: mcp9808: Convert to DTS

Convert mcp9808 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: max44009: Convert to DTS

Convert max44009 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: isl29035: Convert to DTS

Convert isl29035 sensor driver to utilize device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: hp206c: Convert to DTS

Convert hp206c sensor driver to utilize device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: bmm150: Convert to DTS

Convert bmm150 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: bmg160: Convert to DTS

Convert bmg160 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sensor: bmc150_magn: Convert to DTS

Convert bmc150_magn sensor driver to utilize device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

pabigot reviewed (commented):
mcp9808 works (on particle_xenon, didn't test on k64f).  Can't test anything else.

(NB: The driver may other changes to work on Nordic, will submit those once this goes in.)

galak commented on dts/bindings/sensor/semtech,sx9500.yaml:
```diff
@@ -0,0 +1,21 @@
+      type: phandle-array
+      required: false
+
+    reset-gpios:
```
future proofing the binding.  Happy to remove it.

galak commented on dts/bindings/sensor/bosch,bmm150.yaml:
```diff
@@ -0,0 +1,17 @@
+include: i2c-device.yaml
+
+properties:
+    int-gpios:
```
future proofing the binding. Happy to remove it.

carlescufi commented on tests/drivers/build_all/dts_fixup.h:
```diff
@@ -297,6 +297,99 @@
 #define DT_TI_TMP116_BUS_I2C 1
 #endif
 
+#ifndef DT_INST_0_HOPERF_TH02_LABEL
```
Does it still make sense to keep adding those? It’s just for the non-DTS enabled platforms? Couldn’t the dt scripts generate them based on the overall DT files?

galak replied:
It does for now, its the only way we have to ensure a build test for any driver that doesn't have an explicit sample.

The other option is to create a board overlay dts for this test that has all the sensors in it and that we'd keep adding as we have new sensor drivers.

galak replied:
I do like that idea as it gives an example of what the .dts node should look like to someone wanting to add one of the sensors to their boards.

carlescufi reviewed (approved):
Looks good. I left a comment about the fixups but it’s unrelated to this particular PR.

galak commented:
@MaureenHelm I went ahead and remove the extra properties from the bindings since I had to rebase anyways.
